### ISSUE TITLE
run_daily only runs every two days with a trigger at sunset

### ISSUE THREAD

**Issue by EeeTee77** [AppDaemon/appdaemon#1289]
Upgrade to version 4.0.8 seems to have broken the behavior of run_daily triggered with an offset from sunset.

Here is the code used to register the trigger:

```
  def initialize(self):
    self.timers = {}
    self.stateListeners = {}
    self.configureEvent("time_on", True)
    self.configureEvent("time_off", False)

  def configureEvent(self, param, turnOn):
    if param in self.args:
      runAt = self.getTimeSetting(param)
      if turnOn:
        self.timers[param] = self.run_daily(self.turnOn, runAt)
        self.log("Allumage a {}".format(runAt))
      else:
        self.timers[param] = self.run_daily(self.turnOff, runAt)
        self.log("Fermeture a {}".format(runAt))
```

getTimeSetting is only a function that checks whether the script argument is a input_datetime entity or simply a time to parse...

```
Lumieres exterieures:
  module: timer
  class: Timer
  entities: switch.lumiere_exterieur_avant,switch.lumiere_exterieur_garage
  time_on: sunset - 00:30:00
  time_off: sunrise + 00:30:00
```

Before the upgrade, self.turnOn was called everyday. Now, it is called only once per two days. 

The sunrise does not seem affected though; in AppDaemon statistics, I see turnOff being called twice often as turnOn.

Other instances of this automation which are only time based work fine and are called everyday.

**Comment by Justihar** [comment]
EeeTee77,

 

This is the exact issue that myself and some others were having. It is fixed in the current version of development.

 

From: EeeTee77 ***@***.***> 
Sent: Wednesday, June 9, 2021 6:24 AM
To: AppDaemon/appdaemon ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [AppDaemon/appdaemon] run_daily only runs every two days with a trigger at sunset (#1289)

 

Upgrade to version 4.0.8 seems to have broken the behavior of run_daily triggered with an offset from sunset.

Here is the code used to register the trigger:

  def initialize(self):
    self.timers = {}
    self.stateListeners = {}
    self.configureEvent("time_on", True)
    self.configureEvent("time_off", False)
 
  def configureEvent(self, param, turnOn):
    if param in self.args:
      runAt = self.getTimeSetting(param)
      if turnOn:
        self.timers[param] = self.run_daily(self.turnOn, runAt)
        self.log("Allumage a {}".format(runAt))
      else:
        self.timers[param] = self.run_daily(self.turnOff, runAt)
        self.log("Fermeture a {}".format(runAt))

getTimeSetting is only a function that checks whether the script argument is a input_datetime entity or simply a time to parse...

Lumieres exterieures:
  module: timer
  class: Timer
  entities: switch.lumiere_exterieur_avant,switch.lumiere_exterieur_garage
  time_on: sunset - 00:30:00
  time_off: sunrise + 00:30:00

Before the upgrade, self.turnOn was called everyday. Now, it is called only once per two days.

The sunrise does not seem affected though; in AppDaemon statistics, I see turnOff being called twice often as turnOn.

Other instances of this automation which are only time based work fine and are called everyday.

**Comment by EeeTee77** [comment]
I have seen issue #1283 where it can run twice. Is this the one you are referring to? It was stating the callback could be run twice, not that it could be skipped 1 out of 2 times. This is why I logged a new issue...

I can test if the fix solves my issue. Can you point me out the changeset that fixes it?

**Comment by Justihar** [comment]
I think this is the same issue. If I recall all the details correctly, when using an offset, the sunset call is skipped every other day but it doesn't affect everyone, only certain timezones (although, likely based off the lon/lat setting). If you are unable to update to current dev (version 4.1.0), you can try disabling the offset and see if it's still an issue. This will be a temporary fix only if you cannot upgrade to dev until the fix is pushed to a release.

**Comment by ReneTode** [comment]
i can confirm its the same issue.
or at least its fixed by the same fix ;)

**Comment by EeeTee77** [comment]
Then, I will close this issue and wait for the release to be integrated to hassio... 

Thanks for confirming! :)

Great job to all the community by the way!
